client/components/push-notification/settings.jsx

```diff
@@ -0,0 +1,151 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import classNames from 'classnames';
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import Dialog from 'components/dialog';
+import Gridicon from 'components/gridicon';
+import Notice from 'components/notice';
+import observe from 'lib/mixins/data-observe';
+
+const debug = debugFactory( 'calypso:push-notification-settings' );
+
+export default  React.createClass( {
+	displayName: 'PushNotificationSettings',
+
+	mixins: [ observe( 'pushNotifications' ) ],
+
+	propTypes: {
+		pushNotifications: React.PropTypes.object
+	},
+
+	getInitialState: function() {
+		return {
+			showDialog: false
+		};
+	},
+
+	clickHandler: function() {
+		debug( 'Handling click for state %s', this.props.pushNotifications.state );
+
+		if ( 'subscribed' === this.props.pushNotifications.state ) {
+			this.props.pushNotifications.unsubscribe();
+		}
+
+		if ( 'unsubscribed' === this.props.pushNotifications.state ) {
+			this.props.pushNotifications.subscribe();
+		}
+	},
+
+	getBlockedInstruction: function() {
+		return (
+			<Dialog isVisible={ this.state.showDialog } className="push-notification-instruction__dialog" onClose={ this._onCloseDialog }>
+				<div className="push-notification-instruction__content">
+					<div>
+						<div className="push-notification-instruction__title">{ this.translate( 'Enable Browser Notifications' ) }</div>
+						<div className="push-notification-instruction__step">
+							<img height="180px" width="180px" src="/calypso/images/push-notifications/address-bar.svg" />
```

Comment:
I agree with the general idea. I opened the two `svg` files we use here, and I found that even after optimization, inlining them would make the `jsx` file bloated, with a big chunk of "garbage" in the middle of it that would make the code hard to read. I looked at other `svg` files that we already inlined in Calypso, and those are really small, mostly take a single long line in the code, so their presence does not hinder code readability. Unless we can reduce the size of the `svg` text used here I err on the side of not inlining them.